staging/src/k8s.io/client-go/tools/portforward/portforward_test.go

```diff
@@ -165,10 +184,17 @@ func TestParsePortsAndNew(t *testing.T) {
 		// mock-signal the Ready channel
 		close(readyChan)
 
+		// mock random ports being selected for zero local-ports
+		for i := range pf.ports {
+			if pf.ports[i].Local == 0 {
+				pf.ports[i].Local = randomLocalPort
+			}
+		}
```

Comment:
The random ports is mocked in test, so the later check of `reflect.DeepEqual(test.expectedGetPorts, ports)` is merely validating this mock. The code path you fixed with this PR isn't exercised at all, i.e., this test will pass even without your fix.

I would suggest finding a way to exercise the `forward()` method and then validating the local ports is non-zero.

Sorry, I didn't notice this in the previous review.

Reply:
I'm afraid I don't see how the change in tests doesn't exercise the new code path, since removing the `portforward.go` change but keeping the `portforward_test.go` change results in test failure.

I agree that it'd be better for this package's tests to also exercise more internal code. Given that before my change, the end results of `forward()` were already being mocked, and in the interest of keeping this PR focused on the bug fix, would you be ok with merging this as-is? I'd be happy to find a way to improve the tests in a later commit.

Reply:
I take that back. You're absolutely right; I'll see what I can do to exercise `forward()`.

Reply:
@caesarxuchao thanks again for your patience on this PR. Would you mind taking a look at my latest push?